stock_back2back_order_proc/stock.py

```diff
@@ -0,0 +1,552 @@
+                'move_history_ids': False,
+                'move_history_ids2': False,
+                'product_qty': product_qty,
+                'product_uos_qty': product_qty,
+                'picking_id': my_picking_id,
+                'price_unit': move_tbc.price_unit,
+            })
+
+        # Create the move_history_ids (child) if there is.
+        if move_dest_id:
+            move_obj.write(
+                cr, uid, [new_move_id],
+                {'move_history_ids': [(4, move_dest_id)]})
+        return new_move_id
+
+    def update_move_chain_pick(self, cr, uid, move_id, vals, new_picks, context=None):
+        '''Recursively update the new chained move with the new related picking
+            by the first move id until a location in retention mode or the end.
+
+        @return True if ok.
+        '''
+        move_obj = self.pool.get('stock.move')
+        move_tbu = move_obj.browse(cr, uid, move_id, context)
+
+        while True:
+            vals.update(picking_id=new_picks[move_tbu.picking_id.id])
+            move_obj.write(cr, uid, [move_tbu.id], vals, context)
+
+            if not move_tbu.move_dest_id or move_tbu.location_dest_id.retention_mode != 'thru':
+                break
+
+            move_tbu = move_tbu.move_dest_id
+        return True
+
+    def update_move_chain(self, cr, uid, move_id, vals, context=None):
+        '''Recursively update the old chained move by the first move id
+            until a location in retention mode or the end.
+
+        @return True if ok.
+        '''
+        ids = [move_id]
+        move_obj = self.pool.get('stock.move')
+        move_tbu = move_obj.browse(cr, uid, move_id, context)
+
+        while move_tbu.move_dest_id and move_tbu.location_dest_id.retention_mode == 'thru':
+            ids.append(move_tbu.move_dest_id.id)
+            move_tbu = move_tbu.move_dest_id
+
+        move_obj.write(cr, uid, ids, vals, context)
+        return True
+
+    def is_pick_not_empty(self, cr, uid, pick_id, move_obj, context=None):
+        cpt = move_obj.search(
+            cr, uid,
+            [('picking_id', '=', pick_id)],
+            context=context, count=True)
+        return cpt > 0
+
+    def check_production_node_move_chain(
+            self, cr, uid, move_tbc, context=None):
+        if move_tbc.location_id.usage == 'production':
+            return True
+
+        if move_tbc.move_dest_id:
+            if self.check_production_node_move_chain(
+                    cr, uid, move_tbc.move_dest_id, context):
+                return True
+        return False
+
+    def has_production_mode(self, cr, uid, all_moves, context=None):
+        for move in all_moves:
+            if self.check_production_node_move_chain(cr, uid, move, context):
+                return True
+        return False
+
+    # FIXME: needs refactoring, this code is partially duplicated in stock_move.
+    # do_partial()!
+    def do_partial(self, cr, uid, ids, partial_datas, context=None):
```

Comment:
needs fixing: this code does not call the super() implementation of do_partial

* this needs to be noted in the docstrving (and in the module documentation)
* I assume most of the code below is copied from the original implementaiton. Please highlight the changed parts with notes in comments.